packages/client/src/sync/fetcher/trienodefetcher.ts

```diff
@@ -0,0 +1,563 @@
+import { BranchNode, ExtensionNode, LeafNode, Trie, decodeNode } from '@ethereumjs/trie'
+import {
+  Account,
+  KECCAK256_NULL,
+  KECCAK256_RLP,
+  bytesToNibbles,
+  hexToKeybytes,
+  nibblesToCompactBytes,
+  toBytes,
+} from '@ethereumjs/util'
+import { debug as createDebugLogger } from 'debug'
+import { keccak256 } from 'ethereum-cryptography/keccak'
+import { bytesToHex, equalsBytes, hexToBytes } from 'ethereum-cryptography/utils'
+import { OrderedMap } from 'js-sdsl'
+import path from 'path'
+
+import { Fetcher } from './fetcher'
+
+import type { Peer } from '../../net/peer'
+import type { AccountData } from '../../net/protocol'
+import type { FetcherOptions } from './fetcher'
+import type { Job } from './types'
+import type { Nibbles } from '@ethereumjs/trie'
+import type { BatchDBOp } from '@ethereumjs/util'
+import type { Debugger } from 'debug'
+
+const util = require('node:util')
+
+function nibblesToBytes(arr: Nibbles): Uint8Array {
+  const buf = new Uint8Array(arr.length / 2)
+  for (let i = 0; i < buf.length; i++) {
+    let q = i * 2
+    buf[i] = (arr[q] << 4) + arr[++q]
+  }
+
+  return buf
+}
+
+function nibbleToBytes2(arr: Nibbles): Uint8Array {
```

Comment:
This function takes a Nibble array and returns a Uint8Array, packing each two nibbles in a single Uint8Array byte element.